lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +722,805 @@ impl EnsoLexer {
+            let offset       = self.offset.consume();
+            let expr_segment = if is_closed {
+                Token::TextSegmentInterpolate(expr_tokens.into(),offset)
+            } else {
+                Token::TextSegmentUnclosedInterpolate(expr_tokens.into(),offset)
+            };
+            self.text_state.append_segment(expr_segment);
+            self.pop_tokens();
+            self.offset.pop();
+            self.pop_state();
+        } else {
+            self.on_unrecognized(reader);
+        }
+    }
+
+    /// Triggered when encountering an EOF inside an interpolation.
+    fn on_eof_in_interpolate<R:LazyReader>(&mut self, reader:&mut R) {
+        trace!(self.logger,"Text::on_eof_in_interpolate");
+        let interpolate_is_closed = false;
+        self.on_interpolate_end(interpolate_is_closed,reader);
+        let mut text = self.text_state.unsafe_end_literal();
+        text.style   = match text.style {
+            Some(token::TextStyle::FormatLine) => Some(token::TextStyle::UnclosedLine),
+            _                                  => text.style,
+        };
+        let token = text.into();
+        self.append_token(token);
+        self.on_eof(reader);
+    }
+
+    /// Triggered when encountering a newline inside an interpolation.
+    fn on_newline_in_interpolate<R:LazyReader>
+    ( &mut self
+      , line_ending : token::LineEnding
+      , reader      : &mut R
+    ) {
+        trace!(self.logger,"Text::on_newline_in_interpolate");
+        let interpolate_is_closed = false;
+        self.on_interpolate_end(interpolate_is_closed,reader);
+        let current_style = self.text_state.unsafe_current().style.expect("Must be set.");
+        match current_style {
+            token::TextStyle::FormatLine => {
+                let mut text = self.text_state.unsafe_end_literal();
+                text.style   = match text.style {
+                    Some(token::TextStyle::FormatLine) => Some(token::TextStyle::UnclosedLine),
+                    _ => text.style,
+                };
+                let token = text.into();
+                self.append_token(token);
+                self.pop_state();
+                self.block_on_newline(line_ending,reader);
+            },
+            token::TextStyle::FormatInlineBlock => {
+                self.text_inline_block_on_newline(line_ending,reader);
+            },
+            token::TextStyle::FormatBlock => {
+                self.text_on_end_of_line(line_ending,reader);
+            },
+            _ => unreachable_panic!("To reach here the lexer must be inside a format literal."),
+        }
+    }
+
+    /// Triggered when encountering a doc comment inside an interpolate.
+    fn on_doc_comment_in_interpolate<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::submit_plain_segment");
+        let offset     = self.offset.consume();
+        let token      = Token::Unrecognized("##",offset);
```

Comment:
hardcoded literal - what if we change doc comment sequence? Why we have to change it in 2 or more places?